### ISSUE TITLE
Warning being displayed when building in a new git repository with no commits

### ISSUE THREAD

**Issue by atournayre** [docker/buildx#1692]
### Contributing guidelines

- [X] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

The `docker compose build` command end with a warning

### Expected behaviour

No warning and successful build while running `docker compose build` command

### Actual behaviour

The `docker compose build` command end with a warning

### Buildx version

github.com/docker/buildx v0.10.2 00ed17d

### Docker info

```markdown
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.16.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
  scan: Docker Scan (Docker Inc.)
    Version:  v0.23.0
    Path:     /usr/libexec/docker/cli-plugins/docker-scan

Server:
 Containers: 6
  Running: 2
  Paused: 0
  Stopped: 4
 Images: 1230
 Server Version: 23.0.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2456e983eb9e37e47538f59ea18f2043c9a73640
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
 Kernel Version: 5.15.0-67-generic
 Operating System: Ubuntu 20.04.6 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.51GiB
 Name: atournayre
 ID: XK5K:MTZD:I7VI:SHLK:U6PU:QPUP:SGEA:CERO:INTY:5ECE:C3PO:EHGT
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```
```


### Builders list

```markdown
NAME/NODE DRIVER/ENDPOINT STATUS  BUILDKIT PLATFORMS
default * docker                           
  default default         running 23.0.1   linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386 
```
```


### Configuration

```markdown
#syntax=docker/dockerfile:1.4

# The different stages of this Dockerfile are meant to be built into separate images
# https://docs.docker.com/develop/develop-images/multistage-build/#stop-at-a-specific-build-stage
# https://docs.docker.com/compose/compose-file/#target

# Builder images
FROM composer/composer:2-bin AS composer

FROM mlocati/php-extension-installer:latest AS php_extension_installer

# Build Caddy with the Mercure and Vulcain modules
FROM caddy:2.6-builder-alpine AS app_caddy_builder

RUN xcaddy build \
	--with github.com/dunglas/mercure \
	--with github.com/dunglas/mercure/caddy \
	--with github.com/dunglas/vulcain \
	--with github.com/dunglas/vulcain/caddy


# Caddy image
FROM caddy:2.6-alpine AS app_caddy

WORKDIR /srv/app

COPY --from=app_caddy_builder --link /usr/bin/caddy /usr/bin/caddy
COPY --from=app_php --link /srv/app/public public/
COPY --link docker/caddy/Caddyfile /etc/caddy/Caddyfile



Full dockerfile here https://github.com/dunglas/symfony-docker/blob/8b82b74f52dc977eda946f2f78f6bb0d9fc5a2bf/Dockerfile
```


### Logs

```markdown
=> [app_caddy 5/5] COPY --link docker/caddy/Caddyfile /etc/caddy/Caddyfile                                                                                                                                                    0.7s
 => => merging                                                                                                                                                                                                                 0.7s
 => exporting to image                                                                                                                                                                                                         0.9s
 => => exporting layers                                                                                                                                                                                                        0.9s
 => => writing image sha256:507602c94889fc1c45202fcf02d480fd0fffaadf839aae9e2255af408ab73832                                                                                                                                   0.0s
 => => naming to docker.io/library/my-project-caddy                                                                                                                                                                  0.0s
WARN[0118] buildx: failed to get git commit: fatal: argument 'HEAD' ambigu : révision inconnue ou chemin inexistant.
Utilisez '--' pour séparer les chemins des révisions, comme ceci :
'git <commande> [<révision>...] -- [<chemin>...]'
```
```


### Additional info

_No response_

**Comment by atournayre** [comment]
Thanks to @ciaranmcnulty it was solved by adding a dummy commit to the repo.

**Comment by crazy-max** [comment]
> github.com/docker/buildx v0.10.2 [00ed17d](https://github.com/docker/buildx/commit/00ed17df6d20f3ca4553d45789264cdb78506e5f)

This has been fixed in Buildx [0.10.3](https://github.com/docker/buildx/releases/tag/v0.10.3). Please update to latest stable.

**Comment by ciaranmcnulty** [comment]
I encouraged @atournayre to report this but now I realise it's a dupe of #1587 - my apologies!
